internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go

```diff
@@ -77,6 +77,18 @@ func resourceSiteRecoveryProtectionContainerMapping() *pluginsdk.Resource {
 				ValidateFunc:     azure.ValidateResourceID,
 				DiffSuppressFunc: suppress.CaseDifference,
 			},
+			"automatic_update_extensions_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"automation_account_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"automatic_update_extensions_enabled"},
```

Comment:
I believe something weird will happen here if they pass in an `automation_account_id` and have `automatic_update_extensions_enabled` set to false. We should just have this wrapped in a TypeList to show that these two attributes go together.